prov/usnic/src/usdf_av.c

```diff
@@ -506,27 +543,42 @@ static int usdf_av_insertsvc(struct fid_av *fav, const char *node,
 		void *context)
 {
 	struct sockaddr_in *addr;
+	struct usdf_av *av;
+	struct fi_info *info;
 	int ret;
+	bool addr_format_str;
 
 	USDF_TRACE_SYS(AV, "\n");
 
+	av = av_ftou(fav);
+	info = av->av_domain->dom_info;
+	addr_format_str = (info->addr_format == FI_ADDR_STR);
+
 	if (!fav)
 		return -FI_EINVAL;
 
-	addr = calloc(1, sizeof(*addr));
-	if (!addr) {
-		USDF_DBG("address allocation failed\n");
-		return -FI_ENOMEM;
-	}
+	if (addr_format_str) {
+		/* string format should not come with service param. */
+		if (service)
+			return -FI_EINVAL;
 
-	ret = usdf_resolve_addr(node, service, addr);
-	if (ret)
-		goto fail;
+		ret = fav->ops->insert(fav, &node, 1, fi_addr, flags, context);
+	} else {
+		addr = calloc(1, sizeof(*addr));
```

Comment:
In a follow up with some testing, I think you can make this a stack variable. I know it was already dynamically allocated, but I don't think it needs to be.